### PR TITLE
PLT-235 Implement outgoing webhooks.

### PR DESCRIPTION
**Pull request by jwilander**

### CODE REVIEW

File: api/post.go [hunk 1]
```diff
@@ -55,11 +56,15 @@ func createPost(c *Context, w http.ResponseWriter, r *http.Request) {
 
 		return
 	} else {
+		if result := <-Srv.Store.Channel().UpdateLastViewedAt(post.ChannelId, c.Session.UserId); result.Err != nil {
```
Comment (coreyhulen) [inline api/post.go]: why is this moved out?  isn't CreatePost called from a lot of other places?

Comment (jwilander) [inline api/post.go]: This `createPost` not `CreatePost` so it's only called when the API for creating a post is hit

Comment (coreyhulen) [inline api/post.go]: I think you'll get unread action messages for yourself now?  stuff like "Corey joined this channel".  Not a big deal.

**Comment by coreyhulen** [comment]
+1

File: api/post.go (hunk 1 above)
Comment (hmhealey) [inline api/post.go]: No they won't mark channels as unread. Joining and leaving channels just calls CreatePost directly and bypasses createPost so it won't hit this function at all.

**Comment by hmhealey** [comment]
Looks good
